app/helpers/language_helper.py

```diff
@@ -13,17 +14,21 @@
 }
 
 
-def handle_language() -> None:
+def handle_language(metadata: Optional[Mapping[str, Any]] = None) -> None:
     session_store = get_session_store()
+
     if session_store and (session_data := session_store.session_data):
-        launch_language = session_data.launch_language_code or DEFAULT_LANGUAGE_CODE
-        # pylint: disable=assigning-non-slot
-        g.allowed_languages = get_allowed_languages(
-            session_data.schema_name, launch_language
+        metadata = metadata or get_metadata(current_user) or {}
+        schema_name = metadata.get("schema_name")
+
+        launch_language = (
+            session_store.session_data.language_code or DEFAULT_LANGUAGE_CODE
```

Comment:
This needs to get `language_code` from metadata, as session data one is the current language code not necessarily the launch one.